websites/C/Crowdin/presence.ts

```diff
@@ -3,96 +3,124 @@ const presence = new Presence({
 	}),
 	browsingTimestamp = Math.round(Date.now() / 1000);
 
-let translatePageTitle: HTMLElement | null,
-	translatingFile: HTMLElement,
-	translateProject: HTMLElement,
-	translatingLanguage: HTMLElement,
-	profileName: HTMLElement,
-	profileNickname: HTMLElement | null;
+async function getLanguageName(id: string) {
+	const languages = await fetch(
+		"https://api.crowdin.com/api/v2/languages?limit=500",
+		{
+			headers: { Accept: "application/json" }
+		}
+	)
+		.then(
+			r =>
+				r.json() as Promise<{
+					data: { data: CrowdinLanguage }[];
+					pagination: { limit: number; offset: number };
+				}>
+		)
+		.then(r => r.data.map(l => l.data));
+	return languages.find(l => l.id === id)?.name;
+}
 
-presence.on("UpdateData", () => {
+presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
 			details: "Unknown page",
 			largeImageKey: "crowdin",
 			startTimestamp: browsingTimestamp
 		},
-		{ pathname, host } = document.location;
+		{ pathname, host, href } = document.location,
+		pathnameSplit = pathname.split("/"),
+		[showManager, showConversations] = await Promise.all([
+			presence.getSetting<boolean>("showManager"),
+			presence.getSetting<boolean>("showConversations")
+		]);
 
 	switch (host) {
 		case "support.crowdin.com": {
+			const isEnterprise = !!~pathnameSplit.indexOf("enterprise");
 			if (!pathname || pathname === "/")
 				presenceData.details = "On the main support page";
 			else if (
 				pathname.includes("/api/v2/") ||
 				pathname.includes("/enterprise/api/")
 			) {
-				const activeLabel = Array.from(document.querySelectorAll("label")).find(
-					c => c.className?.includes("active")
+				const activeLabel = [...document.querySelectorAll("label")].find(c =>
+					c.className?.includes("active")
 				);
-				presenceData.details = "Reading more about API V2";
-				presenceData.state =
-					activeLabel?.children.length >= 2
-						? activeLabel?.children[1].textContent
-							? activeLabel?.children[1].textContent
-							: activeLabel?.children[0].textContent
-						: activeLabel?.textContent;
+
+				presenceData.details = `Reading the ${
+					isEnterprise ? "Enterprise " : ""
+				}API documentation`;
+
+				presenceData.state = [...(activeLabel?.children ?? [])]
```

Comment:
so if its empty dus it return null/undefined or empty string

Reply:
it maps an empty array and joins is thus making it an empty string

Reply:
do we allow empty strings as state? I thought we said you must just delete presencedata.state then

Reply:
Deleting and having an empty string produces the same result in the end. Doing it like this is more practical, what we usually don't allow is explicitly setting the state or any other property to an empty string. If it's just a possible outcome of a larger code block I think it's fine

Reply:
> do we allow empty strings as state? I thought we said you must just delete presencedata.state then

This is correct

Reply:
there's no point in deleting here since I'm not explicitly setting it to an empty string. The end result is the same, performance wise it would be worse to delete it